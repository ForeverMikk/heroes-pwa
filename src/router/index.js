
export * from './AppRouter';
export * from './PrivatreRoute';
export * from './PublicRoute';