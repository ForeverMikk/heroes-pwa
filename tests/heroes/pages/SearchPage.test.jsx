import { MemoryRouter } from "react-router-dom";
import { render, screen } from '@testing-library/react';
import { SearchPage } from "../../../src/heroes";

/* eslint-disable no-undef */
describe('pruebas en <SearchPage />', () => { 

    test('debe de mostrarse con valores por defecto', () => { 

        const {container} = render(
            <MemoryRouter>
                <SearchPage />
            </MemoryRouter>
        )
        
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrarse a batman y el input con el valor del query', () => { 

        render(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <SearchPage />
            </MemoryRouter>
        )
        
        const input = screen.getByRole('textbox'); //DEfinimos nuestro input
        expect(input.value).toBe('batman') ; //Vemos si en el input se inserto batman
        
        const img = screen.getByRole('img'); //definimos la imagen
        expect(img.src).toContain('/assets/heroes/dc-batman.jpg'); //vemos si se renderiza la imagen del batman

        const alertDanger = screen.getByLabelText('alert-danger');
        expect(alertDanger.style.display).toBe('none');
    });

});