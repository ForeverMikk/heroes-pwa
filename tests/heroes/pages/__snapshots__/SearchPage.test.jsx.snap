// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <SearchPage /> debe de mostrarse con valores por defecto 1`] = `
<div>
  <h1>
    SearchPage
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a Hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search a Hero
      </div>
      <div
        aria-label="alert-danger"
        class="alert alert-danger animate__animated animate__fadeInLeft"
        style="display: none;"
      >
        not found 
        <b />
      </div>
    </div>
  </div>
</div>
`;
